image.pollinations.ai/cloudflare-cache/src/analytics.js

```diff
@@ -84,32 +84,45 @@ export async function sendToAnalytics(request, name, params = {}, env) {
                      '';
     const userAgent = request.headers.get('user-agent') || '';
     const language = request.headers.get('accept-language') || '';
-    const clientIP = getClientIp(request);
+    const clientIP = getClientIp(request) || '::1';
     
     // Process query parameters into safeParams format
     const safeParams = {};
     for (const [key, value] of url.searchParams.entries()) {
       safeParams[key] = value;
     }
     
-    // Build analytics metadata
-    const analyticsMetadata = createAnalyticsMetadata(request, {
-      originalPrompt,
-      safeParams,
-      referrer,
-      ...params
-    });
+    // Extract specific parameters from query params or params object
+    const width = safeParams.width || params.width || 1024;
+    const height = safeParams.height || params.height || 1024;
+    const seed = safeParams.seed || params.seed || 42;
+    const model = safeParams.model || params.model || 'flux';
+    const negative_prompt = safeParams.negative_prompt || params.negative_prompt || 'worst quality, blurry';
     
-    // Build the payload for Google Analytics
+    // Build the payload in the exact same format as the curl command
     const payload = {
-      client_id: clientIP || 'unknown',
-      "events": [{
-        "name": name,
-        "params": {
-          referrer,
-          userAgent,
-          language,
-          ...analyticsMetadata,
+      client_id: clientIP,
+      events: [{
+        name: name,
+        params: {
+          userAgent: userAgent,
+          language: language,
+          width: width,
+          height: height,
+          seed: seed,
+          model: model,
```

Comment:
@ElliotEtag this logic should be much simpler. there is no need to extract each parameter and give defaults and so on here.

Reply:
it should be possible to pass the params object without extracting the fields one by one. potentially adding more data in